### PR TITLE
KLUGE - Option to avoid building RRTMK due to GNU problem

### PR DESCRIPTION
**Pull request by davegill**
TYPE: enhancement

KEYWORDS: GNU, RRTMK

SOURCE: internal

DESCRIPTION OF CHANGES:

Problem:
The GNU compiler versions from 6.* through 8.* issue an internal compiler error on
various RRTMG-based schemes.

Solution:
A modification was introduced in the WRF build to allow the RRTMG-fast version
to skip being built (commit 17b5ac4062010642 with PR #517 "KLUGE: RRTMG FAST +
GNU > 6.3.0 is an internal compiler error"). This same strategy is now used for
the RRTMK scheme.

On a good note ... it looks like GNU verion 9.1.* and later fixes this trouble.

LIST OF MODIFIED FILES:
modified:   arch/postamble
modified:   phys/module_physics_init.F
modified:   phys/module_ra_rrtmg_lwk.F
modified:   phys/module_ra_rrtmg_swk.F
modified:   phys/module_radiation_driver.F
modified:   share/module_check_a_mundo.F

TESTS CONDUCTED:
1. Code is bit-for-bit when not using the RRTMGK scheme with compiling using the default
settings vs when ifdef'ing out the RRTMGK schemes.
2. When no RRTMGK scheme is built, but the scheme is requested, the code successfully traps
the error condition.
```
 &physics
 physics_suite                       = 'CONUS'
 ra_lw_physics                       = 14,    -1,    -1,
 ra_sw_physics                       = 14,    -1,    -1,
```
```
--- ERROR: RRTMG-based KIAPS schemes must be built with a default compile-time flag
--- ERROR: Run ./clean -a, ./configure, ./compile scripts again
-------------- FATAL CALLED ---------------
```

### CODE REVIEW

**Comment by davegill** [comment]
@smileMchen 
Ming,
Does this fix your troubles with the FSBM?

**Comment by smileMchen** [comment]
@davegill 
Dave,
I will rebuild the code and let you know. 
Ming

File: share/module_check_a_mundo.F [hunk 1]
```diff
@@ -2229,6 +2229,22 @@ END FUNCTION bep_bem_nbui_max
...
 
+!-----------------------------------------------------------------------
+!  If the RRTMG KOREA schemes are requested, check that the code with
+!  built to use them.
+!-----------------------------------------------------------------------
+
+#if( BUILD_RRTMK != 1)
+      IF ( ( model_config_rec % ra_lw_physics(1) .EQ. RRTMK_LWSCHEME_FAST )  .OR. &
+           ( model_config_rec % ra_sw_physics(1) .EQ. RRTMK_SWSCHEME_FAST )  ) THEN
+         wrf_err_message = '--- ERROR: RRTM Korea schemes must be built with a default compile-time flag'
```
Comment (weiwangncar) [inline share/module_check_a_mundo.F]: @davegill Maybe KIAPS scheme, not Korean scheme?

File: share/module_check_a_mundo.F [hunk 2]
```diff
@@ -2229,6 +2229,22 @@ END FUNCTION bep_bem_nbui_max
...
 #endif
 
+!-----------------------------------------------------------------------
+!  If the RRTMG KOREA schemes are requested, check that the code with
+!  built to use them.
+!-----------------------------------------------------------------------
+
+#if( BUILD_RRTMK != 1)
+      IF ( ( model_config_rec % ra_lw_physics(1) .EQ. RRTMK_LWSCHEME_FAST )  .OR. &
+           ( model_config_rec % ra_sw_physics(1) .EQ. RRTMK_SWSCHEME_FAST )  ) THEN
```
Comment (weiwangncar) [inline share/module_check_a_mundo.F]: @davegill A pair of eagle eyes also found that it should just be RRTMK_LWSCHEME and RRTMK_SWSCHEME, not _FAST at the end..

File: share/module_check_a_mundo.F (hunk 1 above)
Comment (davegill) [reply share/module_check_a_mundo.F]: Done

File: share/module_check_a_mundo.F (hunk 2 above)
Comment (davegill) [reply share/module_check_a_mundo.F]: Good catch, done
🦅 👁 👍 🥇 🎉

**Comment by liujake** [comment]
@davegill Somehow when I merged this one, I cannot even compile WRF. I use the default gnu/8.3.0 and mpt/2.19. WRF compile Error message:
/work/liuz/REGTEST/develop/wrftladj -I/glade/work/liuz/REGTEST/develop/chem -I/glade/work/liuz/REGTEST/develop/inc -I/glade/u/apps/ch/opt/netcdf/4.7.3/gnu/8.3.0/include    module_ra_rrtmg_swk.f90
module_ra_rrtmg_swk.f90:3265:0:

    use rrsw_kg20_k, only : absa, ka, absb, kb, forref, selfref,                  &

internal compiler error: in gfc_trans_use_stmts, at fortran/trans-decl.c:5122
0x73dfa4 gfc_trans_use_stmts

**Pull request merged**